compiler/semexprs.nim

```diff
@@ -396,12 +396,21 @@ proc isOpImpl(c: PContext, n: PNode, flags: TExprFlags): PNode =
     else:
       res = false
   else:
-    maybeLiftType(t2, c, n.info)
+    if t1.skipTypes({tyAlias}).kind != tyGenericBody:
```

Comment:
Incomprehensible. Why only skip ``tyAlias``? Why only for not tyGenericBody?

I know these are not your faults, but the compiler is full of junk like that. What is the spec that we're following here?

Reply:
this was the simplest patch I found that fixed #13066; `tyGenericBody`, `tyAlias` occur from aliasing a generic head; if I remove this line change, it fails here:
```nim
Error: unhandled exception: ttypetraits.nim(131, 12) `Foo2 is Foo`  [AssertionError]
```